src/main/java/build/buildfarm/instance/memory/MemoryInstance.java

```diff
@@ -450,18 +462,15 @@ private boolean satisfiesRequirements(Platform platform, Operation operation) {
   @Override
   public boolean watchOperation(
       String operationName,
-      boolean watchInitialState,
       Predicate<Operation> watcher) {
-    if (watchInitialState) {
-      Operation operation = getOperation(operationName);
-      if (!watcher.test(operation)) {
-        // watcher processed completed state
-        return true;
-      }
-      if (operation == null || operation.getDone()) {
-        // watcher did not process completed state
-        return false;
-      }
+    Operation operation = getOperation(operationName);
+    if (!watcher.test(operation)) {
+      // watcher processed completed state
+      return true;
+    }
+    if (operation == null || operation.getDone()) {
```

Comment:
How can this even happen? I see the watcher returns falls for these cases.

Reply:
I preferred to be decoupled from any watcher implementation, in particular one that would be unconcerned with the null-ness or done-ness of the operation it is watching. In those cases, I can do nothing further in this function.